src/CMakeLists.txt

```diff
@@ -1332,3 +1337,11 @@ add_custom_target(cephfs_testing DEPENDS
     cephfs-journal-tool
     cephfs-data-scan
     cephfs-table-tool)
+
+if (IS_DIRECTORY "${PROJECT_SOURCE_DIR}/.git")
+  add_custom_target(
+    git-update
+    COMMAND git submodule sync
+    COMMAND git submodule update --force --init --recursive
+    WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}")
+endif()
```

Comment:
i was thinking about 

* using `ExternalProject_Add()`[1] to include the submodules, by overriding its Update/Patch Step so it runs "git submodule update" for a certain submodule when ".git" exists,
* and running "git submodule sync" automatically if ".git" exists

but it is little bit more complicated. if you think it's a viable approach, maybe we can go in this direction.

--
[1] https://cmake.org/cmake/help/v3.0/module/ExternalProject.html